compiler/dio-hdf5/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+nnas_find_package(HDF5 COMPONENTS STATIC QUIET)
+
+if(NOT HDF5_FOUND)
+  message(STATUS "Build dio_hdf5: FAILED (missing HDF5)")
+  return()
+endif(NOT HDF5_FOUND)
+
+file(GLOB_RECURSE SOURCES "src/*.cpp")
+file(GLOB_RECURSE TESTS "src/*.test.cpp")
+list(REMOVE_ITEM SOURCES ${TESTS})
+
+add_library(dio_hdf5 SHARED ${SOURCES})
+target_include_directories(dio_hdf5 PUBLIC include)
+target_include_directories(dio_hdf5 PRIVATE ${HDF5_INCLUDE_DIRS})
+target_link_libraries(dio_hdf5 ${HDF5_CXX_LIBRARIES})
+target_link_libraries(dio_hdf5 loco)
```

Comment:
we can set explict `PUBLIC` or `PRIVATE` for link. I think it would be `PUBLIC`.

Reply:
I think PRIVATE would be better, as we don't want to expose loco interface when linking dio_hdf5.

Reply:
loco is used by HDF5Importer.h, so it should be public.